File: src/api/java/com/brittank88/adinfinitum/api/rarity/ExtendedRarity.java
package com.brittank88.adinfinitum.api.rarity;

import net.minecraft.util.Formatting;

/**
 * An enum providing further rarities than vanilla Minecraft's {@link net.minecraft.util.Rarity Rarity} enum.
 *
 * @author Brittank88
 */
public enum ExtendedRarity {
    /** Common rarity. Present in vanilla. */
    COMMON   (Formatting.WHITE       ),
    /** Uncommon rarity. Present in vanilla. */
    UNCOMMON (Formatting.YELLOW      ),
    /** Rare rarity. Present in vanilla. */
    RARE     (Formatting.AQUA        ),
    /** Epic rarity. Present in vanilla. */
    EPIC     (Formatting.LIGHT_PURPLE),
    /** Legendary rarity. NOT present in vanilla. */
    LEGENDARY(Formatting.GOLD        );

    /** Each {@link ExtendedRarity} has a corresponding formatting that is applied to the {@link net.minecraft.item.Item Item}'s name. */
    public final Formatting formatting;

    // TODO: Implement
    ExtendedRarity(Formatting formatting) { this.formatting = formatting; }
}

File: src/api/java/com/brittank88/adinfinitum/api/rarity/ExtendedRarity.kt
package com.brittank88.adinfinitum.api.rarity

import net.minecraft.util.Formatting

// TODO: Implement

/**
 * An enum providing further rarities than vanilla Minecraft's [Rarity][net.minecraft.util.Rarity] enum.
 *
 * @param value The corresponding formatting that is applied to the [Item][net.minecraft.item.Item]'s name.
 * @author Brittank88
 */
enum class ExtendedRarity(private val value: Formatting) {

    /** Common rarity. Present in vanilla. */
    COMMON(Formatting.WHITE),

    /** Uncommon rarity. Present in vanilla. */
    UNCOMMON(Formatting.YELLOW),

    /** Rare rarity. Present in vanilla. */
    RARE(Formatting.AQUA),

    /** Epic rarity. Present in vanilla. */
    EPIC(Formatting.LIGHT_PURPLE),

    /** Legendary rarity. NOT present in vanilla. */
    LEGENDARY(Formatting.GOLD);
}